docs/chain-abstraction/omnibridge/implementation.md

```diff
@@ -0,0 +1,257 @@
+    TokenCheck -->|Bridged Token| Burn[Burn Token]
+    
+    Lock --> MPCSign[Request MPC Signature]
+    Burn --> MPCSign
+    
+    MPCSign --> Chain{Destination Chain}
+    
+    Chain -->|Ethereum| EVMBridge[EVM Bridge Contract]
+    Chain -->|Bitcoin| BTCBridge[Bitcoin Script]
+    Chain -->|Other| WormBridge[Wormhole Bridge]
+    
+    EVMBridge --> Mint[Mint/Release Tokens]
+    BTCBridge --> Mint
+    WormBridge --> Mint
+    
+    Mint --> End[Transfer Complete]
+```
+
+### Transfer Process
+
+Let's follow what happens when a user wants to transfer tokens from NEAR to another chain:
+
+#### 1. Initiation
+
+The user starts by calling the token contract with:
+
+* Amount to transfer
+* Destination chain and address
+* Fee preferences (whether to pay fees in the token being transferred or in NEAR)
+
+#### 2. Token Lock
+
+The token contract transfers tokens to the locker contract, which:
+
+* Validates the transfer message
+* Assigns a unique nonce
+* Records the pending transfer
+* Emits a transfer event
+
+#### 3. MPC Signing
+
+The bridge contract:
+
+* Requests signature generation
+* MPC nodes jointly generate and aggregate signature
+* Maintains threshold security throughout process
+
+#### 4. Destination Chain
+
+The Bridge Token Factory on the destination chain:
+
+* Verifies the MPC signature
+* Mints equivalent tokens
+* Fees are minted on NEAR side for relayers
+
+---
+
+## Transaction Flow: Other Chains to NEAR
+
+The reverse flow varies based on the source chain:
+
+### 1. Ethereum
+
+Uses NEAR light client for maximum security:
+
+* Burning tokens on source chain
+* Submitting proof to NEAR
+* Verifying proof through light client
+* Releasing tokens to recipient
+
+### 2. Solana
+
+Currently using Wormhole for:
+
+* Message passing between chains
+* Transaction verification
+* Integration with NEAR token factory system
+
+### 3. Other Chains
+
+Initially using Wormhole for:
+
+* Message passing between chains
+* Transaction verification
+* Will transition to Chain Signatures
+
+---
+
+## Security Model
+
+### Trust Assumptions
+
+Omni Bridge requires different trust assumptions depending on the chain connection:
+
+#### For Chain Signatures:
+
+* NEAR Protocol security (2/3+ honest validators)
+* MPC network security (2/3+ honest nodes)
+* No single entity controls enough MPC nodes to forge signatures
+* Correct implementation of the signing protocol
+
+#### For Ethereum/Bitcoin connections:
+
+* Light client security
+* Finality assumptions (e.g., sufficient block confirmations)
+* Chain-specific consensus assumptions
+
+#### For interim Wormhole connections:
+
+* Wormhole Guardian network security
+* We acknowledge this is a temporary trust assumption until Chain Signatures integration is complete
+
+---
+
+## Relayer Network
+
+Relayers are permissionless infrastructure operators who monitor for bridge events and execute cross-chain transactions. Unlike many bridge designs, our relayers cannot:
```

Comment:
Could we explicitly say something like "Relayers do not create additional security assumptions"?